src/System.Diagnostics.Process/src/System/Diagnostics/ProcessStartInfo.cs

```diff
@@ -79,16 +89,47 @@ public bool CreateNoWindow
             set { _createNoWindow = value; }
         }
 
+        public StringDictionary EnvironmentVariables 
+        {
+            get 
+            {
+                                
+                if (_environmentVariables == null)
+                {
+#if PLATFORM_UNIX
+                    _environmentVariables = new StringDictionaryWithComparer(StringComparer.Ordinal);
+#else
+                    _environmentVariables = new StringDictionaryWithComparer(StringComparer.OrdinalIgnoreCase);
+#endif // PLATFORM_UNIX
```

Comment:
This is not how we've been doing PALs in corefx.  There is already a ProcessStartInfo.Unix.cs and a ProcessStartInfo.Windows.cs file; things specific to one or the other need to go there, with common code here.  You can see elsewhere in this file how we handle this exact case:
https://github.com/safern/corefx/blob/06070a85a0f6fd7abfa79bebbb9b23b673768159/src/System.Diagnostics.Process/src/System/Diagnostics/ProcessStartInfo.cs#L134
and we should just do the same thing here (you can see that CaseSensitiveEnvironmentVariables is defined differently in each of those .Unix.cs and .Windows.cs files).